dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/constant/DefaultConstantService.java

```diff
@@ -106,7 +140,8 @@ public Constant getConstant( String uid )
     @Transactional( readOnly = true )
     public List<Constant> getAllConstants()
     {
-        return constantStore.getAll();
+        return (List<Constant>) ALL_CONSTANTS_CACHE.get( ALL_CONSTANTS_KEY, key -> constantStore.getAll() )
+            .orElse( new ArrayList<Constant>() );
```

Comment:
Is there a reason not to use `Collections.emptyList()` instead?

Reply:
No reason. Thanks for the tip!